app/src/main/java/org/mozilla/focus/topsites/TopSites.kt

```diff
@@ -0,0 +1,221 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.topsites
+
+import androidx.compose.foundation.ExperimentalFoundationApi
+import androidx.compose.foundation.background
+import androidx.compose.foundation.combinedClickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Card
+import androidx.compose.material.DropdownMenu
+import androidx.compose.material.DropdownMenuItem
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import mozilla.components.feature.top.sites.TopSite
+import mozilla.components.support.ktx.kotlin.getRepresentativeCharacter
+import org.mozilla.focus.R
+
+/**
+ * A list of top sites.
+ *
+ * @param topSites List of [TopSite] to display.
+ * @param menuItems List of [TopSiteMenuItem] to display in a top site dropdown menu.
+ * @param onTopSiteClick Invoked when the user clicks on a top site.
+ */
+@Composable
+fun TopSites(
+    topSites: List<TopSite>,
+    menuItems: List<TopSiteMenuItem>,
+    onTopSiteClick: (TopSite) -> Unit = {}
+) {
+    Row(
+        modifier = Modifier
+            .padding(horizontal = 10.dp)
+            .size(width = 324.dp, height = 84.dp),
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        Row(
```

Comment:
Is there a reason for having a Row inside another row?

Reply:
https://www.youtube.com/watch?v=6cqbSNERnc0 😂